code/game/objects/items/weapons/storage/storage.dm

```diff
@@ -56,8 +61,48 @@
 						return
 					M.put_in_l_hand(src)
 			add_fingerprint(usr)
-			return
 
+//Proc for Contents transfer to other containers, bins, turf, and more possible additions.
+/obj/item/weapon/storage/proc/contentto(obj/dest_object, mob/user)
+	playsound(loc, "rustle", 50, 1, -5)
+	switch( storage_contents_dump_act(dest_object) )

```

Comment:
Instead of returning a number in the children procs, and then using that number to decide what code to run, just put that code IN the children procs. Byond will then automatically pick the correct version of the proc for the correct type, no typechecking needed!